app/controllers/orders/client_summary.js

```diff
@@ -19,7 +23,60 @@ export default detail.extend({
     return this.get("store").peekAll("identity_type");
   }),
 
+  isLongDescription: Ember.computed("model.purposeDescription", function() {
+    return this.get("model.purposeDescription.length") > LONG_DESCRIPTION_SIZE;
+  }),
+
+  toggleDisplayShowMore: Ember.observer("model.purposeDescription", function() {
+    const isLongDescription = this.get("isLongDescription");
+    isLongDescription
+      ? this.set("displayShowMore", true)
+      : this.set("displayShowMore", false);
+  }),
+
+  shortPurposeDescription: Ember.computed(
+    "showInlineEdit",
+    "displayShowMore",
+    function() {
+      let purposeDescription = this.get("model.purposeDescription");
```

Comment:
if one day you need to re-use this logic, it'd be cool to have it as a helper

e.g `{{ellipsis-text maxChar=170 collapsed=displayShowMore text=purposeDescription }}`

Not a change request, just a thought : )